### PR TITLE
Feature/agent/new design apis

### PR DESCRIPTION
**Pull request by kpango**
### Description:
v0 component internal logic changes for V1 api
this PR resolves thease comment
- https://github.com/vdaas/vald/pull/742#issuecomment-708206178
- https://github.com/vdaas/vald/pull/742#discussion_r504433961

### Environment:

- Go Version: 1.15.2
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [x] Bug fix [type/bug]
- [x] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by vankichi — APPROVED** [review]
LGTM

File: pkg/agent/core/ngt/handler/grpc/handler.go [hunk 1]
```diff
@@ -21,10 +21,11 @@ import (
...
 	"fmt"
 	"strconv"
+	"sync"
 
-	agent "github.com/vdaas/vald/apis/grpc/agent/core"
-	"github.com/vdaas/vald/apis/grpc/gateway/vald"
-	"github.com/vdaas/vald/apis/grpc/payload"
+	agent "github.com/vdaas/vald/apis/grpc/v1/agent/core"
+	"github.com/vdaas/vald/apis/grpc/v1/payload"
+	vald "github.com/vdaas/vald/apis/grpc/v1/vald"
```
Comment (kmrmt) [inline pkg/agent/core/ngt/handler/grpc/handler.go:28]: I think this sentence is redundant. Is this on purpose?

Comment (kpango) [reply pkg/agent/core/ngt/handler/grpc/handler.go]: fixed

**Comment by kevindiu** [comment]
@kpango please fix ci error
```

# github.com/vdaas/vald/internal/client
internal/client/client.go:36:30: undefined: payload.Backup_Vector
internal/client/client.go:40:30: undefined: payload.Backup_Vector
# github.com/vdaas/vald/internal/client/compressor
internal/client/compressor/client.go:32:48: undefined: payload.Backup_Vector
internal/client/compressor/client.go:34:37: undefined: payload.Backup_Vector
internal/client/compressor/client.go:35:46: undefined: payload.Backup_Vectors
internal/client/compressor/client.go:62:68: undefined: payload.Backup_Vector
internal/client/compressor/client.go:91:53: undefined: payload.Backup_Vector
internal/client/compressor/client.go:103:62: undefined: payload.Backup_Vectors
# github.com/tensorflow/tensorflow/tensorflow/go
/usr/bin/ld: /usr/local/lib/libtensorflow.so: .dynsym local symbol at index 851 (>= sh_info of 2)
/usr/bin/ld: /usr/local/lib/libtensorflow.so: .dynsym local symbol at index 1959 (>= sh_info of 2)
/usr/bin/ld: /usr/local/lib/libtensorflow.so: .dynsym local symbol at index 2329 (>= sh_info of 2)
/usr/bin/ld: /usr/local/lib/libtensorflow.so: .dynsym local symbol at index 2517 (>= sh_info of 2)
/usr/bin/ld: /usr/local/lib/libtensorflow.so: .dynsym local symbol at index 2732 (>= sh_info of 2)
/usr/bin/ld: /usr/local/lib/libtensorflow.so: .dynsym local symbol at index 2733 (>= sh_info of 2)
```

**Comment by kpango** [comment]
I don't care about test code and hack dir, I'll fix it @ #740

**Comment by kpango** [comment]
all docker image builds are passed.

**Review by kmrmt — APPROVED** [review]

**Comment by kevindiu** [comment]
I think this error occurs before the test case is even executed. 
there is no `backup_vector` definition here. https://github.com/vdaas/vald/blob/1aeed5c4eb3880bbfdc05f8984a0ec52c9e8bb83/apis/grpc/payload/payload.pb.go

**Review by kevindiu — APPROVED** [review]
LGTM

**Pull request merged**